### PR TITLE
Optimize CI a bit

### PR DESCRIPTION
**Pull request by Goooler**
## Description

Speed up build on CI and housekeeping.

#### Automatic tests

- [ ] Added test cases
  
#### Manual tests
- [x] Done  
  

#### Build tasks success  

Successfully running following tasks on local:
- [x] `./gradlew assembledebug`
- [x] `./gradlew spotlessCheck`

### CODE REVIEW

File: .github/workflows/android-build.yml [hunk 1]
```diff
@@ -15,27 +15,28 @@ jobs:
...
+        uses: actions/setup-java@v3
         with:
           distribution: "temurin"
           java-version: 11
-          cache: 'gradle'
       - name: Check formatting using spotless
-        run: ./gradlew spotlessCheck --stacktrace
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: spotlessCheck --stacktrace
```
Comment (Goooler) [inline .github/workflows/android-build.yml:27]: https://github.com/gradle/gradle-build-action

Ref https://github.com/detekt/detekt/pull/3723, it's really so fast to run jobs, and many other advantages.

File: .github/workflows/android-main.yml [hunk 2]
```diff
@@ -63,19 +69,15 @@ jobs:
...
       - name: Gradle cache
+        uses: gradle/gradle-build-action@v2
+      - name: AVD cache
         uses: actions/cache@v2
-        with:
-          path: |
-            ~/.gradle/caches
-            ~/.gradle/wrapper
-          key: gradle-${{ runner.os }}-${{ hashFiles('**/*.gradle*') }}-${{ hashFiles('**/gradle/wrapper/gradle-wrapper.properties') }}-${{ hashFiles('**/buildSrc/**/*.kt') }}
-      - uses: actions/cache@v2
```
Comment (Goooler) [inline .github/workflows/android-main.yml:78]: https://github.com/ReactiveCircus/android-emulator-runner#usage

File: .github/workflows/android-build.yml [hunk 3]
```diff
@@ -15,27 +15,28 @@ jobs:
     name: Check spotless
     runs-on: ubuntu-20.04
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - name: Set up JDK 11
-        uses: actions/setup-java@v2
+        uses: actions/setup-java@v3
```
Comment (EmmanuelMess) [inline .github/workflows/android-build.yml:20]: And the cache?

**Comment by EmmanuelMess** [comment]
This runs slower, might we be missing some cache?

**Comment by EmmanuelMess** [comment]
It looks like it run in 18m the second time, not significantly faster than older cached runs. Any ideas?

**Comment by Goooler** [comment]
https://github.com/TeamAmaze/AmazeFileManager/runs/5480556993?check_suite_focus=true

```
BUILD SUCCESSFUL in 2m 35s
130 actionable tasks: 130 executed
```

VS
https://github.com/TeamAmaze/AmazeFileManager/runs/5439915461?check_suite_focus=true
```
BUILD SUCCESSFUL in 4m 32s
130 actionable tasks: 130 executed
```

**Comment by Goooler** [comment]
https://github.com/TeamAmaze/AmazeFileManager/runs/5480144746?check_suite_focus=true
```
BUILD SUCCESSFUL in 19m 40s
67 actionable tasks: 14 executed, 53 up-to-date
Watching 722 directories to track changes
Created user preferences directory.
```

https://github.com/TeamAmaze/AmazeFileManager/actions/runs/1956602475
```
BUILD SUCCESSFUL in 14m 56s
67 actionable tasks: 14 executed, 53 up-to-date
Watching 722 directories to track changes
Created user preferences directory.
```

**Comment by EmmanuelMess** [comment]
See https://github.com/TeamAmaze/AmazeFileManager/runs/5473395277?check_suite_focus=true vs https://github.com/TeamAmaze/AmazeFileManager/runs/5480556993?check_suite_focus=true .

**Review by EmmanuelMess — APPROVED** [review]

**Review by VishalNehra — APPROVED** [review]

**Pull request merged**